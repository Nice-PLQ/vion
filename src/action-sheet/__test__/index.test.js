import { mount } from '@vue/test-utils';
import ActionSheet from '../action-sheet.vue';
import VPopup from '../../popup/popup.vue';

describe('ActionSheet', () => {
  const wrapper = mount(ActionSheet, {
    props: {
      items: ['选项一', '选项二', { text: '删除', color: 'red' }],
      title: '标题',
      showCancel: true,
      cancelText: 'cancel',
      showSelectIcon: true,
      teleportDisabled: true
    },
    global: {
      components: {
        VPopup
      }
    }
  });

  test('选项数量', () => {
    expect(wrapper.find('.v-action-sheet').element).toMatchSnapshot();
    expect(wrapper.findAll('.v-action-sheet__menu').length).toBe(3);
  });

  test('显示与隐藏', async () => {
    await wrapper.setProps({ visible: true });
    expect(wrapper.find('.v-popup__mask').isVisible()).toBe(true);
    expect(wrapper.find('.v-popup__content').isVisible()).toBe(true);
  });

  test('标题', () => {
    expect(wrapper.find('.v-action-sheet__title').text()).toBe('标题');
  });

  test('显示取消', async () => {
    const cancel = wrapper.find('.v-action-sheet__action');
    expect(cancel.text()).toBe('cancel');
    await cancel.trigger('click');
    expect(wrapper.emitted('cancel')).toBeTruthy();
  });

  test('关闭', async () => {
    await wrapper.find('.v-popup__mask').trigger('click');
    expect(wrapper.emitted('close')).toBeTruthy();
  });
});
