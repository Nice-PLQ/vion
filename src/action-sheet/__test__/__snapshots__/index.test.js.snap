// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionSheet 选项数量 1`] = `
<div
  aria-modal="true"
  class="v-action-sheet"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      aria-label="关闭"
      class="v-popup__mask"
      role="button"
      style="display: none;"
    />
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="false"
  >
    <!--v-if-->
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="slide-bottom"
    persisted="true"
  >
    <div
      class="v-popup__content v-popup__content_bottom"
      style="display: none;"
    >
      
      <div
        class="v-action-sheet__panel"
      >
        <div
          class="v-action-sheet__title"
        >
          
          <p
            class="v-action-sheet__title-text"
          >
            标题
          </p>
          
        </div>
        
        <div
          aria-label=""
          class="v-action-sheet__menu"
          role="button"
        >
          <div
            aria-hidden="false"
            class="v-action-sheet__cell"
          >
            
            <div
              class="v-action-sheet__cell-text"
            >
              选项一
            </div>
            
            <!--v-if-->
          </div>
        </div>
        <div
          aria-label=""
          class="v-action-sheet__menu"
          role="button"
        >
          <div
            aria-hidden="false"
            class="v-action-sheet__cell"
          >
            
            <div
              class="v-action-sheet__cell-text"
            >
              选项二
            </div>
            
            <!--v-if-->
          </div>
        </div>
        <div
          aria-label=""
          class="v-action-sheet__menu"
          role="button"
        >
          <div
            aria-hidden="false"
            class="v-action-sheet__cell"
          >
            
            <div
              class="v-action-sheet__cell-text"
              style="color: red;"
            >
              删除
            </div>
            
            <!--v-if-->
          </div>
        </div>
        
        <div
          class="v-action-sheet__action v-action-sheet__safe-area"
          role="button"
        >
          <div
            class="v-action-sheet__cell"
          >
            cancel
          </div>
        </div>
      </div>
      
    </div>
  </transition-stub>
</div>
`;
